src/awk.h

```diff
@@ -73,9 +78,15 @@ typedef struct awk_rule_
   struct awk_rule_ *next_rule;    /* linked list */
   enum {BEGIN,BEGIN_REC,END_REC,END,REGEXP} ruletype;
   const char *pattern;        /* pcre pattern string */
-  const u_char *tables;       /* pcre NLS tables */
-  pcre *re;                   /* pcre compiled pattern */
-  pcre_extra *pe;             /* pcre optimized pattern */
+  #ifdef XASTIR_LEGACY_PCRE
+    const u_char *tables;       /* pcre NLS tables */
+    pcre *re;                   /* pcre compiled pattern */
+    pcre_extra *pe;             /* pcre optimized pattern */
+  #else
+    const uint8_t *tables;       /* pcre2 NLS tables */
+    pcre2_code *re;             /* pcre2 compiled pattern */
+    void *pe;                   /* pcre2 doesn't use separate optimization */
```

Comment:
I didn't see this being used in the PCRE2 cases, do we need to define it elsewhere?

Reply:
Yes, we need to define "*pe" in the header even though it's never used, because there is an initializer in testdbfawk.h that tries to define a default set of rules, and it uses C initializer format --- so there's a value for pe in it (null).  If we don't have a place holder in the PCRE2 case, then we have to have two sets of initializers, one for PCRE2 and once for legacy PCRE.

If we remove PCRE support and go with ONLY PCRE2, we can remove "pe" from the struct *and* change the initializers so they don't try to initialize that element.